modules/oe_content_featured_media_field/src/Plugin/Field/FieldType/FeaturedMediaFieldItem.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_featured_media_field\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem;
+use Drupal\Core\Field\PreconfiguredFieldUiOptionsInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\OptionsProviderInterface;
+
+/**
+ * Plugin implementation of the 'oe_featured_media' field type.
+ *
+ * @FieldType(
+ *   id = "oe_featured_media",
+ *   label = @Translation("Featured media"),
+ *   module = "oe_content_featured_media_field",
+ *   category = @Translation("OpenEuropa"),
+ *   description = @Translation("Stores a featured media item and caption."),
+ *   default_formatter = "oe_featured_media_label",
+ *   default_widget = "oe_featured_media_widget",
+ *   column_groups = {
+ *     "target_id" = {
+ *       "label" = @Translation("Media item"),
+ *       "translatable" = FALSE
+ *     },
+ *     "caption" = {
+ *       "label" = @Translation("Caption"),
+ *       "translatable" = TRUE
+ *     },
+ *   },
+ *   list_class = "\Drupal\Core\Field\EntityReferenceFieldItemList"
+ * )
+ */
+class FeaturedMediaFieldItem extends EntityReferenceItem implements OptionsProviderInterface, PreconfiguredFieldUiOptionsInterface {
+
```

Comment:
Add isEmpty() method to check the caption field.

Reply:
We consider the field empty if there is no referenced entity. We don't need to look for the caption field.

Reply:
It's a field type with 2 items and we ignore one of them but I can save the node having values only in caption and then the value is not even saved.

Reply:
We take care with #states to make the Media item required.